### PR TITLE
Snow 2.0.1

### PR DESCRIPTION
**Pull request by weizman**
> This PR merge action was migrated to https://github.com/MetaMask/metamask-extension/pull/22195 (see [comment](https://github.com/MetaMask/metamask-extension/pull/20471#issuecomment-1845284816))

This is an attempt to bump Snow in MetaMask from 1.5.0 to [2.0.1](https://github.com/LavaMoat/snow/pull/134).

In between Snow went through a lot. The major things to be aware of are:

* For better security, Snow will start blocking some core browser actions that aren't normally used.
* For better security, Snow expects some minimal level of CSP protection for help.
* For better security, Snow expects to be loaded and called on every HTML file served by the same origin.

### QA Help

This PR might need some QA. This introduces zero new features, only hardens security of the product.
Meaning, there isn't anything new to focus on here in regards of QA. Instead, a general QA should be taken against the app, to see that the security hardening introduced does not harm the normal already-existing functionality of the MM app.

### CODE REVIEW

File: app/trezor-usb-permissions.html [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <!DOCTYPE html>
 <html lang="en">
 <head>
+    <script src="./snow.js" type="text/javascript"  charset="utf-8"></script>
+    <script src="./use-snow.js" type="text/javascript" charset="utf-8"></script>
```
Comment (Gudahtt) [inline app/trezor-usb-permissions.html:5]: I'm not sure how the Trezor integration uses this page. We'll want to test with Trezor to ensure this hasn't broken anything.

File: app/manifest/v2/chrome.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "content_security_policy": "frame-ancestors 'none'; script-src 'self' 'wasm-unsafe-eval'; object-src 'self'",
+  "content_security_policy": "frame-ancestors 'none'; script-src 'self' 'wasm-unsafe-eval'; object-src 'none'",
```
Comment (Gudahtt) [inline app/manifest/v2/chrome.json:2]: I'm not aware of anyplace we're using `<object>` or `<embed>` elements, so this is probably a safe change. I was the one who set `object-src` to `self` in the first place. My motivation for doing that was just that it matched the browser default CSP.

**Review by Gudahtt — DISMISSED** [review]
LGTM! Though I haven't manually tested this at all yet.

**Review by digiwand — DISMISSED** [review]
👍🏼  confirmed I was able to sign a `personal_sign` message using the Trezor Model T


comparing Snow 1.5.0 → Snow 2.0.1 
https://github.com/LavaMoat/snow/compare/916da6c...08c242d

**Comment by Gudahtt** [comment]
Added the platform team label as well because that label check doesn't recognize the lavamoat team yet

File: app/trezor-usb-permissions.html (hunk 1 above)
Comment (weizman) [reply app/trezor-usb-permissions.html:5]: https://github.com/MetaMask/metamask-extension/pull/20471#pullrequestreview-1583103931

**Comment by chloeYue** [comment]
@weizman Hello Gal, QA will tackle this PR very soon, could you please provide more information for manual testing steps ? It will be very helpful, thanks!

**Comment by weizman** [comment]
Thank you @chloeYue , do you have any questions on top of the already existing description?

(see https://github.com/MetaMask/metamask-extension/pull/20471#issue-1852605939 section `QA Help`)

**Comment by chloeYue** [comment]
Hello @weizman , QA passed for this PR. I also added the label.

**Review by digiwand — APPROVED** [review]

**Comment by weizman** [comment]
This PR merge action was migrated to https://github.com/MetaMask/metamask-extension/pull/22195 (see https://github.com/MetaMask/metamask-extension/pull/20471#issuecomment-1845284816)

**Pull request closed without merging**